Fix/ios run simulator

matteocrippa opened the pull request:
@tvolkert I noticed the test was wrong and once fixed the regex was not valid to fix the issue, I managed to fix both the test and regex now (old PR #19863)

matteocrippa committed:
upgraded test

Looking better in the dump, seems that the ESC ascii is attach to url

matteocrippa committed:
fix regex to avoid to collect again ]ESC

matteocrippa commented:
seems all fine except for ipv6, but I'm not that good in regex.
open in any suggestion to improve that

tvolkert commented on packages/flutter_tools/lib/src/protocol_discovery.dart:
```diff
@@ -60,7 +60,7 @@ class ProtocolDiscovery {
     Uri uri;
 
-    final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on (http://[^ \n]+)');
+    final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on ((http:|[/][/]|www.)([a-z]|[A-Z]|[:0-9]|[/.])*)');
```
It should never be `www`, and you don't need square brackets around the individual slashes, which would make the first part `(http:|//)`

In the second part, You can just put all the acceptable characters in one "or" group, like so: `[a-zA-Z0-9:/\.]`

Finally, you want to require something in the second part, so you'd use `+` and not `*`

That would make the whole thing:

```dart
final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on ((http|//)[a-zA-Z0-9:/\.]+)');
```

tvolkert replied:
Ah, we need to add the open and close brackets, and apparently equals to the list of legal characters:

```dart
[a-zA-Z0-9:/=\.\[\]]
```

matteocrippa replied:
Dart was not working, I added a couple of escape more and it seems to work with this one:
`((http|\/\/)[a-zA-Z0-9:/=\.\\[\\]]+)`

tvolkert commented on packages/flutter_tools/test/protocol_discovery_test.dart:
```diff
@@ -68,7 +68,7 @@ void main() {
       testUsingContext('discovers uri even if logs has ESC Ascii', () async {
         initialize();
-        logReader.addLine('Observatory listening on http://127.0.0.1:3333 \x1b[');
+        logReader.addLine('Observatory listening on http://127.0.0.1:3333\x1b[');
```
Presumably the changes above are also made to work on Windows, so let's add some tests of both IPv6 and Windows URLs with the trailing `\x1b[` sequence

matteocrippa committed:
fix regex for ipv6

matteocrippa committed:
add test for ipv6 with Esc Ascii

matteocrippa committed:
Update protocol_discovery_test.dart

tvolkert commented on packages/flutter_tools/test/protocol_discovery_test.dart:
```diff
@@ -195,6 +195,26 @@ void main() {
+
+        // Get next port future.
+        final Future<Uri> nextUri = discoverer.uri;
+        logReader.addLine('I/flutter : Observatory listening on http://127.0.0.1:54804/PTwjm8Ii8qg=/\x1b[');
```
That's not an IPv6 address

matteocrippa replied:
I'll replace with `http://[::1]:54777/PTwjm8Ii8qg=/` but is not that clear for me how the test above works.
It does the same and it's used for ipv6, I haven't digged if flag `ipv6: true` do some mock replace

matteocrippa committed:
test with ipv6 address

not pretty sure it works in that manner or mock somehow replace that thanks to `ipv6` trigger in _ProtocolDiscovery_

tvolkert commented:
What's the current status of this PR?

matteocrippa committed:
fix lint test

tvolkert commented:
Tests still failing

Commit:
fix linting for test protocol discovery

Commit:
fix lint issue raised by CI test

matteocrippa commented:
seems everything ok except for win, but I didn't catch the error there in the tests

Hixie commented on packages/flutter_tools/test/protocol_discovery_test.dart:
```diff
@@ -48,7 +48,8 @@ void main() {
 
-      testUsingContext('discovers uri if logs already produced output', () async {
+      testUsingContext('discovers uri if logs already produced output',
+          () async {
```
nit: probably more readable if you leave it unwrapped as before

Hixie commented on packages/flutter_tools/test/protocol_discovery_test.dart:
```diff
@@ -66,9 +68,10 @@ void main() {
         initialize();
-        logReader.addLine('Observatory listening on http://127.0.0.1:3333 \x1b[');
+        logReader
+            .addLine('Observatory listening on http://127.0.0.1:3333\x1b[');
```
nit: ditto

Hixie commented on packages/flutter_tools/test/protocol_discovery_test.dart:
```diff
@@ -88,13 +91,16 @@ void main() {
         final Uri actualUri = await uriFuture.timeout(
-            const Duration(milliseconds: 100), onTimeout: () => timeoutUri);
+            const Duration(milliseconds: 100),
+            onTimeout: () => timeoutUri);
```
nit: probably more readable if you put a comma after `timeoutUri` and then the `)` on the next line.

in general, i'm guessing you ran `dartfmt`. That's fine, but you may need to do some hand-tweaking after to avoid some of the less readable output it produces. Generally I find it works best with a very wide line wrapping option.

Hixie commented:
@matteocrippa Looks like the tests are passing now, what's the plan here?

Commit:
fix linting/indentation

matteocrippa commented:
@Hixie I should have manually fix the indentation issue of Vscode, if it is ok and works think we can merge so we close the issue adding also extra tests

tvolkert reviewed (approved):
LGTM
